solutions/swb-reference/integration-tests/tests/multiStep/sshKeys.test.ts

```diff
@@ -69,63 +70,112 @@ describe('sshKeys API multiStep integration test', () => {
   });
 
   describe('negative test', () => {
-    // zipping corresponding resources for testing
-    const testBundle = [
-      {
-        userName1: 'projectAdmin1',
-        userName2: 'researcher1',
-        session1: () => pa1Session,
-        session2: () => rs1Session,
-        projectId: () => project1Id
-      },
-      {
-        userName1: 'researcher1',
-        userName2: 'projectAdmin1',
-        session1: () => rs1Session,
-        session2: () => pa1Session,
-        projectId: () => project1Id
-      }
-    ];
-
-    test.each(testBundle)('within project boundary', async (testCase) => {
-      const {
-        userName1,
-        userName2,
-        session1: session1func,
-        session2: session2Func,
-        projectId: projectIdFunc
-      } = testCase;
-      const session1 = session1func();
-      const session2 = session2Func();
-      const projectId = projectIdFunc();
-
-      console.log(`${userName2} cannot manage ${userName1}'s key within ${projectId}`);
-
-      console.log(`${userName1} creating a key`);
-      const { data: createdSshKey } = await session1.resources.projects.project(projectId).sshKeys().create();
-
-      console.log(`${userName2} fails to retrieve ${userName1}'s key by listUserSshKeysForProject`);
-      const { data: listedSshKeyAfterDelete } = await session2.resources.projects
-        .project(projectId)
-        .sshKeys()
-        .get();
-      expect(listedSshKeyAfterDelete.sshKeys.length).toEqual(0);
-
-      console.log(`${userName2} fails to delete the ${userName1}'s key`);
-      try {
-        await session2.resources.projects.project(projectId).sshKeys().sshKey(createdSshKey.id).purge();
-      } catch (e) {
-        checkHttpError(
-          e,
-          new HttpError(403, {
-            error: 'Forbidden',
-            message: `Current user ${session2.getUserId()} cannot delete a key they do not own`
-          })
-        );
-      }
-
-      console.log(`${userName1} can hard delete their own key`);
-      await session1.resources.projects.project(projectId).sshKeys().sshKey(createdSshKey.id).purge();
+    describe('within project boundary', () => {
+      // zipping corresponding resources for testing
+      const testBundle = [
+        {
+          userName1: 'projectAdmin1',
+          userName2: 'researcher1',
+          session1: () => pa1Session,
```

Comment:
Why does it need to be function? Can we pass in pa1Session directly?

Reply:
it will evaluated to undefined if not using it as a function..

Reply:
Emm is it because pa1Session is initialized in beforeAll function? Maybe we can put it in beforeAll as well to avoid it. No need to change it now, just curious.